HyperVExtension/src/HyperVExtension/Models/VmCredentialAdaptiveCardSession.cs

```diff
@@ -129,63 +196,22 @@ private void Dispose(bool disposing)
         }
     }
 
-    private string GetTemplate()
+    private string LoadTemplate()
     {
-        return Resources.ReplaceIdentifers(_credentialUITemplate, Resources.GetHyperVResourceIdentifiers(), Logging.Logger());
-    }
-
-    private static readonly string _credentialUITemplate = @"
-{
-    ""$schema"": ""http://adaptivecards.io/schemas/adaptive-card.json"",
-    ""type"": ""AdaptiveCard"",
-    ""version"": ""1.5"",
-    ""body"": [
-        {
-            ""type"": ""TextBlock"",
-            ""text"": ""%VmCredentialRequest/Title%"",
-            ""horizontalAlignment"": ""Center"",
-            ""wrap"": true,
-            ""style"": ""heading""
-        },
+        if (!string.IsNullOrEmpty(_template))
         {
-            ""type"": ""TextBlock"",
-            ""text"": ""%VmCredentialRequest/Description1%"",
-            ""wrap"": true
-        },
-        {
-            ""type"": ""TextBlock"",
-            ""text"": ""%VmCredentialRequest/Description2%"",
-            ""wrap"": true
-        },
-        {
-            ""type"": ""Input.Text"",
-            ""id"": ""UserVal"",
-            ""label"": ""%VmCredentialRequest/UsernameLabel%"",
-            ""isRequired"": true,
-            ""errorMessage"": ""%VmCredentialRequest/UsernameErrorMsg%""
-        },
-        {
-            ""type"": ""Input.Text"",
-            ""id"": ""PassVal"",
-            ""style"": ""Password"",
-            ""label"": ""%VmCredentialRequest/PasswordLabel%""
-        }
-    ],
-    ""actions"": [
-        {
-            ""type"": ""Action.Execute"",
-            ""title"": ""%VmCredentialRequest/OkText%"",
-            ""id"": ""okAction"",
-            ""data"": {
-                ""id"": ""okAction""
-            }
-        },
-        {
-            ""type"": ""Action.Execute"",
-            ""title"": ""%VmCredentialRequest/CancelText%"",
-            ""id"": ""cancelAction""
+            return _template;
         }
-    ]
-}
-";
+
+        var path = Path.Combine(AppContext.BaseDirectory, @"HyperVExtension\Templates\", "VmCredentialAdaptiveCardTemplate.json");
+        _template = File.ReadAllText(path, Encoding.Default) ?? throw new FileNotFoundException(path);

```

Comment:
Will this exception crash DevHome?

Reply:
It will crash Hyper-V extension.

Reply:
Will it negatively affect DevHome if the extension crashes?  Won't the reference to the OOP object become invalid if this happens?

As long as DevHome won't crash if it uses an extension that is crashed I'm okay.

Reply:
If Dev Home cannot handle an extension's crash, it's Dev Home's problem. Any extension can potentially crash and it's Dev Home's responsibility to handle it.

Here it's failfast most likely indicating a broken setup or a developer's error. We want to get a dump and investigate and fix the cause of the exception.